scalalib/api/src/mill/scalalib/api/ZincWorkerUtil.scala

```diff
@@ -68,7 +68,7 @@ trait ZincWorkerUtil {
 
   def scalaJSBinaryVersion(scalaJSVersion: String): String = scalaJSVersion match {
     case _ if scalaJSVersion.startsWith("0.6.") =>
-      throw new Exception("Scala.js 0.6 is not supported")
+      throw Result.Failure("Scala.js 0.6 is not supported")
```

Comment:
How about introducing the discussed `mill.api.ResultFailureException extends MillException` as part of this PR, and throw it instead of `Result.Failure`?

We can add any automatic lifting of it into a `Result.Failure` in a separate PR.